flink-table-store-hive/src/main/java/org/apache/flink/table/store/mapred/TableStoreInputFormat.java

```diff
@@ -91,65 +101,115 @@ public RecordReader<Void, RowDataContainer> getRecordReader(
                 splitLength);
     }
 
-    private FileStoreImpl createFileStore(JobConf jobConf) {
-        TableStoreJobConf wrapper = new TableStoreJobConf(jobConf);
-
-        String dbName = wrapper.getDbName();
-        String tableName = wrapper.getTableName();
-
-        Configuration options = new Configuration();
-        String tableLocation = wrapper.getLocation();
-        wrapper.updateFileStoreOptions(options);
-
-        String user = wrapper.getFileStoreUser();
-
-        List<String> columnNames = wrapper.getColumnNames();
-        List<LogicalType> columnTypes = wrapper.getColumnTypes();
-
-        List<String> partitionColumnNames = wrapper.getPartitionColumnNames();
-
-        RowType rowType =
-                RowType.of(
-                        columnTypes.toArray(new LogicalType[0]),
-                        columnNames.toArray(new String[0]));
-        LogicalType[] partitionLogicalTypes =
-                partitionColumnNames.stream()
-                        .map(s -> columnTypes.get(columnNames.indexOf(s)))
-                        .toArray(LogicalType[]::new);
-        RowType partitionType =
-                RowType.of(partitionLogicalTypes, partitionColumnNames.toArray(new String[0]));
-
-        Optional<List<String>> optionalPrimaryKeyNames = wrapper.getPrimaryKeyNames();
-        if (optionalPrimaryKeyNames.isPresent()) {
-            Function<String, RowType.RowField> rowFieldMapper =
-                    s -> {
-                        int idx = columnNames.indexOf(s);
-                        Preconditions.checkState(
-                                idx >= 0,
-                                "Primary key column "
-                                        + s
-                                        + " not found in table "
-                                        + dbName
-                                        + "."
-                                        + tableName);
-                        return new RowType.RowField(s, columnTypes.get(idx));
-                    };
-            RowType primaryKeyType =
-                    new RowType(
-                            optionalPrimaryKeyNames.get().stream()
-                                    .map(rowFieldMapper)
-                                    .collect(Collectors.toList()));
-            return FileStoreImpl.createWithPrimaryKey(
-                    tableLocation,
-                    new FileStoreOptions(options),
-                    user,
-                    partitionType,
-                    primaryKeyType,
-                    rowType,
-                    FileStoreOptions.MergeEngine.DEDUPLICATE);
-        } else {
-            return FileStoreImpl.createWithValueCount(
-                    tableLocation, new FileStoreOptions(options), user, partitionType, rowType);
+    private static class FileStoreWrapper {
```

Comment:
Maybe we should have a `FileStoreFactory` in `flink-table-store-core` after https://github.com/apache/flink-table-store/pull/101
Then we can create `Predicate` from that. (Maybe get row type from it)